crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -215,32 +218,25 @@ fn merge_stream_and_receiver(
                     }
                 }
 
-                Ok(update) = room_info_notable_update_receiver.recv() => {
-                    // We are temporarily listening to all updates.
-                    /*
-                    use RoomInfoNotableUpdateReasons as NotableUpdate;
-
-                    let reasons = &update.reasons;
-
-                    // We are interested by these _reasons_.
-                    if reasons.contains(NotableUpdate::LATEST_EVENT) ||
-                        reasons.contains(NotableUpdate::RECENCY_STAMP) ||
-                        reasons.contains(NotableUpdate::READ_RECEIPT) ||
-                        reasons.contains(NotableUpdate::UNREAD_MARKER) ||
-                        reasons.contains(NotableUpdate::MEMBERSHIP) {
-                     */
-                        // Emit a `VectorDiff::Set` for the specific rooms.
-                        if let Some(index) = raw_current_values.iter().position(|room| room.room_id() == update.room_id) {
-                            let room = &raw_current_values[index];
-                            let update = VectorDiff::Set { index, value: room.clone() };
-                    /*
-                            trace!(room = %room.room_id(), "updated because of notable reason: {reasons:?}");
-                    */
-                            yield vec![update];
+                update = room_info_notable_update_receiver.recv() => {
+                    match update {
+                        Ok(update) => {
+                            // Emit a `VectorDiff::Set` for the specific rooms.
+                            if let Some(index) = raw_current_values.iter().position(|room| room.room_id() == update.room_id) {
+                                let room = &raw_current_values[index];
+                                let update = VectorDiff::Set { index, value: room.clone() };
+                                yield vec![update];
+                            }
+                        }
+
+                        Err(RecvError::Closed) => {
+                            error!("Cannot receive room info notable updates because the sender has been closed");
+                        }
```

Comment:
hacktualley: if this stream has closed, then this infinite loop will iloop on this case, right? Maybe we should return at this point?

Reply:
I'm adding a little `break` then. Good catch. It would have been caught previously, but not anymore.